### DISCUSSION TITLE
What is the replacement of SPI.setBitOrder()？

### DISCUSSION THREAD

**Discussion by sdwhsrw** [earlephilhower/arduino-pico#1810] [Q&A]
Need this function but it’s marked as deprecated in source code.

**Comment by earlephilhower** [comment]
That function has been [deprecated by the Arduino guys](https://www.arduino.cc/reference/en/language/functions/communication/spi/setbitorder/) for a long time.
> This function should not be used in new projects. Use [SPISettings](https://www.arduino.cc/reference/en/language/functions/communication/spi/spisettings). with [SPI.beginTransaction()](https://www.arduino.cc/reference/en/language/functions/communication/spi/begintransaction). to configure SPI parameters.

The proper way is to pass in the appropriately configured `SPISettings` to `SPI.beginTransaction()` which defines the clock, bit order, and mode.
